src/uct/tcp/tcp_cm.c

```diff
@@ -293,12 +296,21 @@ uct_tcp_ep_t *uct_tcp_cm_get_ep(uct_tcp_iface_t *iface,
     }
 
     ep = ucs_container_of(elem, uct_tcp_ep_t, elem);
-    ucs_assert(ep->flags & (with_ctx_cap | UCT_TCP_EP_FLAG_ON_MATCH_CTX));
+    ucs_assert(ep->flags & UCT_TCP_EP_FLAG_ON_MATCH_CTX);
+
+    if (queue_type == UCS_CONN_MATCH_QUEUE_UNEXP) {
+        ucs_assert(ep->flags & UCT_TCP_EP_FLAG_CTX_TYPE_RX);
+    } else if (!(ep->flags & UCT_TCP_EP_FLAG_CTX_TYPE_TX)) {
+        ucs_assert(ep->flags & UCT_TCP_EP_FLAG_CTX_TYPE_RX);
+    }
```

Comment:
same assert?
may be re-define condition like ```if (statement1 || statement2)``` ?

Reply:
done